### ISSUE TITLE
Enable setting operationId on request

### ISSUE THREAD

**Issue by jonbe242** [RSuter/NSwag#1360]
For logging and statistics purposes it would be great if the generated code could set a property on the HttpRequestMessage containing the operationId of the swagger spec.

This way one could write generic code in a HttpClient DelegatingHandler.

I suspect not all are interested in this, but an option in NSwagStudio to do it when generating code would be great.

**Comment by jonbe242** [comment]
One of the PrepareRequest methods could get the operationId as a parameter, or similar, as an alternate solution.

**Comment by RSuter** [comment]
You can define a base class and generate a create method for the message... see settings

**Comment by jonbe242** [comment]
Yes I can. I can also use PrepareRequest to modify the request after it was created.

But I'd like to get the operationId from the swagger spec, without having to use reflection to reconstruct it from client class name and method.
